pretyping/pretyping.ml

```diff
@@ -186,9 +186,23 @@ type inference_flags = {
 let frozen_holes (sigma, sigma') =
   (); fun ev -> Evar.Map.mem ev (Evd.undefined_map sigma)
 
+let rec advance sigma evk =
+  let evi = Evd.find sigma evk in
+  match evi.evar_body with
+  | Evar_empty -> Some evk
+  | Evar_defined v ->
+      if Option.default false (Store.get evi.evar_extra Evarutil.cleared) then
+        let (evk,_) = Term.destEvar v in
+        advance sigma evk
+      else
+        None
+
 let pending_holes (sigma, sigma') =
```

Comment:
This PR may be a good occasion to add a comment explaining what `pending_holes` is.